### PR TITLE
memcache_proxy: return "App Engine" for "version"

### PR DESCRIPTION
**Pull request by bshaffer**
Some clients use `Memcached::getVersion` to determine if the server is running and the connection is working.  The current implementation [throws an error](https://github.com/GoogleCloudPlatform/appengine-sidecars-docker/blob/master/memcache_proxy/dtog.go#L250), which results in clients making the mistaken assumption the server is not responding.

This PR adds support for the [version](https://github.com/memcached/memcached/blob/1.2.4/doc/protocol.txt#L422) protocol command, but returns "not implemented". This allows the client to verify the connection is successful, but does not return any actual versioning information since we do not actually have a version number.

**Usage**
```
$ telnet <proxy-ip> <proxy-host>
Trying <proxy-ip>...
Connected to <proxy-ip>.
Escape character is '^]'.
version
VERSION not implemented
```

**Alternatives**
 -`VERSION appengine_proxy.v1`
 -`VERSION 1`

Using "not implemented" is the safest/least complicated solution, but either of these alternatives would also be okay.

### CODE REVIEW

File: memcache_proxy/dtog.go [hunk 1]
```diff
@@ -674,3 +676,13 @@ func (s *streams) stats() error {
 	s.out.printLn([]byte("END"))
 	return nil
 }
+
+// version handles the "version" command on the memcached socket and
+// returns "not implemented" as this proxy is not versioned
```
Comment (broady) [inline memcache_proxy/dtog.go]: it returns "VERSION not implemented"

Add a period to the end of the comment.

**Comment by broady** [comment]
Add a test?

**Comment by broady** [comment]
LGTM

**Comment by bshaffer** [comment]
ok, added tests

File: memcache_proxy/dtog.go (hunk 1 above)
Comment (bshaffer) [inline memcache_proxy/dtog.go]: done

**Comment by broady** [comment]
I think this is good for now, but maybe we can figure out something better. How about just "App Engine"?

The spec doesn't say there needs to be a version number. My memcache on ubuntu a string in addition to the version number.
```
$ nc localhost 11211
version
VERSION 1.4.14 (Ubuntu)
```

**Comment by broady** [comment]
LGTM

**Pull request merged**